File: SNT.py
'''
Nhập vào 1 số.

Kiểm tra số đó có phải là số nguyên tố không?
Xuất: true, nếu đó là số nguyên tố, ngược lại xuất false

Ví dụ:

Input	
4
Output
false
'''

n = int(input())
def snt(n):
    if n < 2:
        return False
    for i in range(2, int(n ** 0.5) + 1):
        if n % i == 0:
            return False
    return True
if(snt(n)):
    print("true")
else:
    print("false")

File: LTCB/C++/VongLap/Revise/SNT.cpp
/*
Nhập vào 1 số.

Kiểm tra số đó có phải là số nguyên tố không?
Xuất: true, nếu đó là số nguyên tố, ngược lại xuất false

Ví dụ:

Input	Output
4       false

*/

#include <iostream>
#include <cmath>
using namespace std;

int snt(int n){
    if(n < 2) return 0; // Số nguyên tố nhỏ nhất là 2
    for(int i = 2; i <= sqrt(n); i++){
        if(n % i == 0) return 0; // Nếu n chia hết cho i thì n không phải là số nguyên tố
    }
    return 1;
}
int main(){
    int n;
    cin >> n;
    if(snt(n)) cout << "true";
    else cout << "false";
    return 0;
}